HikVisionSource/src/main/java/io/vantiq/extsrc/HikVisionSource/HikVision.java

```diff
@@ -0,0 +1,1493 @@
+/*
+ * Copyright (c) 2020 Vantiq, Inc.
+ *
+ * All rights reserved.
+ * 
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.HikVisionSource;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import com.sun.jna.NativeLong;
+import com.sun.jna.Pointer;
+import com.sun.jna.ptr.ByteByReference;
+import com.sun.jna.ptr.IntByReference;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vantiq.client.ResponseHandler;
+import io.vantiq.client.Vantiq;
+import io.vantiq.client.VantiqError;
+import io.vantiq.extjsdk.ExtensionWebSocketClient;
+import io.vantiq.extsrc.HikVisionSource.HCNetSDK.FMSGCallBack;
+import io.vantiq.extsrc.HikVisionSource.HCNetSDK.NET_DVR_ALARMER;
+import io.vantiq.extsrc.HikVisionSource.HCNetSDK.NET_DVR_DEVICEINFO_V30;
+import io.vantiq.extsrc.HikVisionSource.HCNetSDK.NET_DVR_SETUPALARM_PARAM;
+import io.vantiq.extsrc.HikVisionSource.exception.VantiqHikVisionException;
+import io.vantiq.extjsdk.ExtensionServiceMessage;
+
+/**
+ * Thie class implememnt the SDK with HikVision Cameras , from one hand it
+ * activate the different cameras and from the other hand it hanldes the
+ * different notifications accpeted by the different cameras.
+ */
+public class HikVision {
+    /*
+     * public class RealDataCallBack implements HCNetSDK.FRealDataCallBack_V30 {
+     * public void invoke(NativeLong lRealHandle, int dwDataType, ByteByReference
+     * pBuffer, int dwBufSize, Pointer pUser) {
+     * System.out.println("Receive notification on class RealDataCallBack");
+     * 
+     * } }
```

Comment:
please remove this & add some java doc on what's going on here...

Need java doc & comments throughout.  Can't really review without.

Reply:
done - hope enough